calculator.c

```diff
@@ -0,0 +1,80 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <string.h>
+
+#define MAX 100
+
+// Complete structure of the code is explained in comments
+
+// Checks if the character is an operator
+int is_operator(char c) {
+    return (c == '+' || c == '-' || c == '*' || c == '/');
+}
+
+// Calculate the expression
+double calculate(const char* expr) {
+    double number = 0;    // Stores the number read from the expression
+    double stack[MAX];    
+    int top = -1;
+    char last_operator = '+';  // Stores the previous operator
+
+    for (int i = 0; i < strlen(expr); i++) {
+        if (isspace(expr[i])) {     // Check if the character is a space
+            continue;
+        } else if (isdigit(expr[i])) {  // Check if the character is a digit (number)
```

Comment:
Write your own implementation for isdigit.

Reply:
okay!!